pkg/sentry/fsimpl/kernfs/filesystem.go

```diff
@@ -768,6 +768,10 @@ func (fs *Filesystem) SymlinkAt(ctx context.Context, rp *vfs.ResolvingPath, targ
 func (fs *Filesystem) UnlinkAt(ctx context.Context, rp *vfs.ResolvingPath) error {
 	fs.mu.Lock()
 	defer fs.mu.Unlock()
+
+	// Store the name before walkExistingLocked as rp will be advanced past the
+	// name in the following call.
+	name := rp.Component()
```

Comment:
I think I saw this in a previous change as well. Is it correct for all fs using kernfs?

Reply:
Yes